### PR TITLE
Init / system-logger containers are not defining the resources to use.

### PR DESCRIPTION
**Pull request by rcosnita**
This PR adds supports for fine grained control over the quotas of each default container currently defined by the cass-operator:

* server-config-init
* cassandra
* server-system-logger

Before this PR, only cassandra container had quota configured. This PR makes the configuration of additional containers explicitly visible in the CRD definition:

```yaml
resourceServerSystemLogger:
    requests:
      memory: 1Gi
      cpu: 1
    limits:
      memory: 1Gi
      cpu: 1
resourceServerConfigInit:
    requests:
      memory: 1Gi
      cpu: 1
    limits:
      memory: 1Gi
      cpu: 1
```

### CODE REVIEW

**Comment by jimdickinson** [comment]
Thank you for the PR! :) Can you add something similar for the reaper sidecar? Also, are you OK with signing a DataStax CLA? We're getting the website for it updated next week, but we can probably merge this in advance if you're up for it.

**Comment by rcosnita** [comment]
@jimdickinson I will update the reaper sidecar by the end of tomorrow. Regarding the CLA, I've just signed it.

File: operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go [hunk 1]
```diff
@@ -162,6 +162,12 @@ type CassandraDatacenterSpec struct {
 	// Kubernetes resource requests and limits, per pod
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 
+	// Kubernetes resource requests and limits per system logger container.
+	ResourceServerSystemLogger corev1.ResourceRequirements `json:"resources-server-system-logger, omitempty"`
+
+	// Kubernetes resource requests and limits per server config initialization container.
+	ResourceServerConfigInit corev1.ResourceRequirements `json:"resources-server-config-init, omitempty"`
```
Comment (jimdickinson) [inline operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go:169]:
```suggestion
	ConfigBuilderResources corev1.ResourceRequirements `json:"configBuilderResources, omitempty"`
```

File: operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go [hunk 2]
```diff
@@ -162,6 +162,12 @@ type CassandraDatacenterSpec struct {
 	// Kubernetes resource requests and limits, per pod
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 
+	// Kubernetes resource requests and limits per system logger container.
+	ResourceServerSystemLogger corev1.ResourceRequirements `json:"resources-server-system-logger, omitempty"`
```
Comment (jimdickinson) [inline operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go:166]:
```suggestion
	SystemLoggerResources corev1.ResourceRequirements `json:"systemLoggerResources, omitempty"`
```

File: operator/pkg/reconciliation/defaults.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package reconciliation
+
+var (
+	// Provides reasonable defaults for the logger container.
+	DefaultsLoggerContainer = buildResourceRequirements(500, 256)
+
+	// Provides reasonable defaults for the configuration container.
+	DefaultsConfigInitContainer = buildResourceRequirements(1000, 256)
+
+	DefaultsReaperContainer = buildResourceRequirements(2000, 512)
```
Comment (jimdickinson) [inline operator/pkg/reconciliation/defaults.go:10]: @jsanda what do you think is reasonable here?

File: operator/pkg/reconciliation/defaults.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package reconciliation
+
+var (
+	// Provides reasonable defaults for the logger container.
+	DefaultsLoggerContainer = buildResourceRequirements(500, 256)
```
Comment (jimdickinson) [inline operator/pkg/reconciliation/defaults.go:5]: pretty sure these could drop to
```suggestion
	DefaultsLoggerContainer = buildResourceRequirements(100, 64)
```

File: operator/pkg/reconciliation/defaults.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+	// Provides reasonable defaults for the configuration container.
+	DefaultsConfigInitContainer = buildResourceRequirements(1000, 256)
```
Comment (jimdickinson) [inline operator/pkg/reconciliation/defaults.go:8]: Nothing hugely CPU bound here, although we should test, too. Too high of a CPU request will make issues for our test and examples running on small setups, though.

```suggestion
	DefaultsConfigInitContainer = buildResourceRequirements(250, 256)
```

**Comment by jimdickinson** [comment]
@rcosnita left some feedback that was a little nitpicky, but I don't think we can merge it as is without the couple small changes. Thank you for the solid effort here! Which CLA did you sign?

**Comment by rcosnita** [comment]
@jimdickinson  Thank you for the review. I will address by the end of tomorrow.  Regarding the CLA, here is a quick screenshot:

Screenshot 2020-08-11 at 10 32 45

File: operator/pkg/reconciliation/defaults.go (hunk 3 above)
Comment (jsanda) [reply operator/pkg/reconciliation/defaults.go:10]: Sorry for the delayed response. I will try to have some numbers for you today.

Comment (rcosnita) [reply operator/pkg/reconciliation/defaults.go:11]: Should we leave the values as is and tune them later? @jsanda / @jimdickinson  what do you think?

**Comment by rcosnita** [comment]
@jimdickinson Thank you for the review. I merged all the changes in the PR.

**Comment by rcosnita** [comment]
Just as a final info here is a valid configuration now supported with the code from the pr:

```yaml
systemLoggerResources:
    requests:
      memory: 1Gi
      cpu: 1
    limits:
      memory: 1Gi
      cpu: 1
configBuilderResources:
    requests:
      memory: 1Gi
      cpu: 1
    limits:
      memory: 1Gi
      cpu: 1
reaper:
  enabled: true
  resources:
    requests:
      memory: 1Gi
      cpu: 1
    limits:
      memory: 1Gi
      cpu: 1
```

**Comment by jimdickinson** [comment]
Looks great! I think I'll add on whatever we need to make these marked as optional on the CRD.

**Pull request merged**